client/signup/config/flows-pure.js

```diff
@@ -297,7 +295,14 @@ export function generateFlows( { getSiteDestination = noop, getPostsDestination
 	if ( config.isEnabled( 'signup/import-landing-handler' ) ) {
 		flows.import = {
 			steps: [ 'from-url', 'user', 'domains' ],
-			destination: ( { siteSlug } ) => `/settings/import/${ siteSlug }`,
+			destination: ( { importSiteDetails, importUrl, siteSlug } ) =>
+				addQueryArgs(
+					{
+						engine: importSiteDetails.engine === 'wix' ? 'wix' : null,
+						'from-site': importUrl && ( encodeURIComponent( importUrl ) || null ),
```

Comment:
Am I right to think that here, when `importUrl` is falsey we get `from-site=false`? And there is a chance that `importUrl` is truthy but `encodeURIComponent( importUrl )` may evaluate to falsey, in which case we want to set `from-site` to `null`?

So 3 things are possible: `false`, a string or `null`.

I think we could instead do `( importUrl && encodeURIComponent( importUrl ) ) || null` to achieve string (result of  encodeURIComponent) or `null`.

I might have confused myself there though, so ffti